noir-projects/noir-contracts/contracts/lending_contract/src/asset.nr

```diff
@@ -13,14 +13,16 @@ struct Asset {
     oracle: AztecAddress,
 }
 
-global SERIALIZED_LEN: Field = 4;
+global SERIALIZED_LEN: Field = 6;
 
 impl Serialize<SERIALIZED_LEN> for Asset {
     fn serialize(Asset: Asset) -> [Field; SERIALIZED_LEN] {
         [
-            Asset.interest_accumulator.to_integer(),
+            Asset.interest_accumulator.lo as Field,
+            Asset.interest_accumulator.hi as Field,
             Asset.last_updated_ts as Field,
-            Asset.loan_to_value.to_integer(),
+            Asset.loan_to_value.lo as Field,
+            Asset.loan_to_value.hi as Field,
```

Comment:
Do you know if before this change the noir would force a too-large integer value to field by wrapping it around modulus?

Reply:
The loan to value is a u128 so should not be the case. This change was mainly made as the serialization/deserialization got a bit messed up 💀